rolluptool/src/classes/RollupServiceTest3.cls

```diff
@@ -528,6 +528,539 @@ private with sharing class RollupServiceTest3
+		rollupSummary.FieldToAggregate__c = aggregateField;
+		rollupSummary.AggregateOperation__c = RollupSummaries.AggregateOperation.Sum.name();
+		rollupSummary.AggregateResultField__c = aggregateResultField;
+		rollupSummary.Active__c = true;
+		rollupSummary.CalculationMode__c = RollupSummaries.CalculationMode.Developer.name();
+		insert rollupSummary;
+
+		// Insert parents
+		SObject parentA = parentType.newSObject();
+		parentA.put('Name', 'ParentA');
+		SObject parentB = parentType.newSObject();
+		parentB.put('Name', 'ParentB');
+		SObject parentC = parentType.newSObject();
+		parentC.put('Name', 'ParentC');
+		List<SObject> parents = new List<SObject> { parentA, parentB, parentC };
+		insert parents;
+
+		// Insert children
+		List<SObject> children = new List<SObject>();
+		for(SObject parent : parents)
+		{
+			String name = (String) parent.get('Name');
+			SObject child1 = childType.newSObject();
+			child1.put(relationshipField, parent.Id);
+			child1.put(aggregateField, 20);
+			children.add(child1);
+			SObject child2 = childType.newSObject();
+			child2.put(relationshipField, parent.Id);
+			child2.put(aggregateField, 20);
+			children.add(child2);
+		}
+		insert children;
+
+		// Assert nothing has changed on db
+		Map<Id, SObject> assertParents = new Map<Id, SObject>(Database.query(String.format('select id, {0} from {1}', new List<String>{ aggregateResultField, parentObjectName })));
+		System.assertEquals(null, (Decimal) assertParents.get(parentA.id).get(aggregateResultField));
+		System.assertEquals(null, (Decimal) assertParents.get(parentB.id).get(aggregateResultField));
+		System.assertEquals(null, (Decimal) assertParents.get(parentC.id).get(aggregateResultField));
+
+		// insert a new child
+		List<SObject> newChildren = new List<SObject>();
+		for(SObject parent : parents)
+		{
+			SObject child3 = childType.newSObject();
+			child3.put(relationshipField, parent.Id);
+			child3.put(aggregateField, 2);
+			newChildren.add(child3);
+		}
+		insert newChildren;
+
+		// Assert nothing has changed on db
+		assertParents = new Map<Id, SObject>(Database.query(String.format('select id, {0} from {1}', new List<String>{ aggregateResultField, parentObjectName })));
+		System.assertEquals(null, (Decimal) assertParents.get(parentA.id).get(aggregateResultField));
+		System.assertEquals(null, (Decimal) assertParents.get(parentB.id).get(aggregateResultField));
+		System.assertEquals(null, (Decimal) assertParents.get(parentC.id).get(aggregateResultField));
+
+		// combine the new and old
+		List<SObject> newAndOldChildren = children.clone();
+		newAndOldChildren.addAll(newChildren);
+
+		// Call developer 'trigger like' API simulating an update
+		RollupService.rollup(new Map<Id, SObject>(children), new Map<Id, SObject>(newAndOldChildren), childType);
+
+		// Assert parents are updated
+		assertParents = new Map<Id, SObject>(Database.query(String.format('select id, {0} from {1}', new List<String>{ aggregateResultField, parentObjectName })));
```

Comment:
Might be worth checking Limits class before and after have not changed (within expected tolerances) for SOQL and DML usage for super paranoid testing? ;-)